docker/nginx/conf.d/client.conf

```diff
@@ -127,10 +128,42 @@ server {
 	}
 
 	location /hns {
-		include /etc/nginx/conf.d/include/cors;
 		include /etc/nginx/conf.d/include/proxy-buffer;
 
-		proxy_pass http://handshake-api:3100;
+		set $skylink ''; # placeholder for the raw 46 bit skylink
+		set $rest ''; # placeholder for the rest of the url that gets appended to skylink (path and args)
+		
+		# resolve handshake domain by requesting to /hnsres endpoint and assign correct values to $skylink and $rest
+		access_by_lua_block {
+			local json = require('cjson')
+			local hns_domain_name, request_uri_rest = string.match(ngx.var.request_uri, "/hns/([^/?]+)(.*)")
+			local hnsres_res = ngx.location.capture("/hnsres/" .. hns_domain_name)
+			local hnsres_json = json.decode(hnsres_res.body)
```

Comment:
You want to handle responses with status != 200 here differently. Lua is debug hell if you don't handle errors properly

Reply:
yeah I should probably 404 here